src/client/components/features/FormationAlternance/Rechercher/RechercherFormationAlternance.tsx

```diff
@@ -23,63 +23,53 @@ import {
 } from '~/client/components/layouts/RechercherSolution/Resultat/ResultatRechercherSolution';
 import { EnTete } from '~/client/components/ui/EnTete/EnTete';
 import { LightHero, LightHeroPrimaryText, LightHeroSecondaryText } from '~/client/components/ui/Hero/LightHero';
-import { useDependency } from '~/client/context/dependenciesContainer.context';
 import { useFormationQuery } from '~/client/hooks/useFormationQuery';
-import { FormationService } from '~/client/services/formation/formation.service';
 import empty from '~/client/utils/empty';
 import { formatRechercherSolutionDocumentTitle } from '~/client/utils/formatRechercherSolutionDocumentTitle.util';
-import { isSuccess } from '~/server/errors/either';
 import { Erreur } from '~/server/errors/erreur.types';
 import { RésultatRechercheFormation } from '~/server/formations/domain/formation';
 import { transformObjectToQueryString } from '~/server/services/utils/urlParams.util';
 
 const PREFIX_TITRE_PAGE = 'Rechercher une formation en apprentissage';
 
-export default function RechercherFormation() {
-	const router = useRouter();
+interface RechercherFormationProps {
+	erreurRecherche?: Erreur
+	resultats?: Array<RésultatRechercheFormation>
+}
```

Comment:
T'as forcément l'un des deux et seulement un des deux nan ? 
```suggestion
type RechercherFormationProps = {
	erreurRecherche?: never
	resultats: Array<RésultatRechercheFormation>
} | {
	erreurRecherche: Erreur
	resultats?: never
}
```